.github/workflows/ci.yml

```diff
@@ -86,6 +86,7 @@ jobs:
       run: |
         set -e
 
+        cargo check
```

Comment:
Can you change this to `cargo metadata --format-version 1 > /dev/null`? `cargo check` may fail for other reasons (ie, there's an issue with the code uploaded in a PR).

Reply:
Hello @joshlf,

I've assessed that both `cargo check` and `cargo metadata` aren't appropriate solutions for addressing this issue. They don't align with the workflow of this job. The root problem lies in the job's failure to pause when a command encounters an error, and then pass that outcome to the subsequent command in the pipeline.

Given this, I am proposing a solution that revolves around adjusting the `set` command. To be more precise, by introducing the `-o pipefail` option, we can make the command proficient at recognizing failures. This, in turn, empowers the `-e` flag to conclude the process. Consequently, the refined `set` command would appear as follows: `set -eo pipefail`. For further elaboration on the `set` command, additional details can be found using `set --help`.

Additionally, I've noticed that other jobs also use `set -e` and command piping, which might have similar issues. Would it be acceptable to apply these changes to all those jobs?

Your guidance on this matter would be greatly appreciated.

Reply:
Yeah, setting `set -eo pipefail` everywhere would be great!